hydromt/models/model_utils.py

```diff
@@ -0,0 +1,144 @@
+"""Utility functions for Model and ModelComponent class."""
+import glob
+import os
+from logging import Logger
+from os.path import basename, dirname, isdir, join
+from tempfile import TemporaryDirectory
+from typing import Dict
+
+import xarray as xr
+
+from hydromt._typing.type_def import DeferedFileClose, StrPath, XArrayDict
+from hydromt.gis.raster import GEO_MAP_COORD
+
+
+def read_nc(
+    fn: StrPath,
+    root,
+    logger: Logger,
+    mask_and_scale: bool = False,
+    single_var_as_array: bool = True,
+    load: bool = False,
+    **kwargs,
+) -> Dict[str, xr.Dataset]:
+    """Read netcdf files at <root>/<fn> and return as dict of xarray.Dataset.
+
+    NOTE: Unless `single_var_as_array` is set to False a single-variable data source
+    will be returned as :py:class:`xarray.DataArray` rather than
+    :py:class:`xarray.Dataset`.
+    key-word arguments are passed to :py:func:`xarray.open_dataset`.
+
+    Parameters
+    ----------
+    fn : str
+        filename relative to model root, may contain wildcards
+    mask_and_scale : bool, optional
+        If True, replace array values equal to _FillValue with NA and scale values
+        according to the formula original_values * scale_factor + add_offset, where
+        _FillValue, scale_factor and add_offset are taken from variable attributes
+        (if they exist).
+    single_var_as_array : bool, optional
+        If True, return a DataArray if the dataset consists of a single variable.
+        If False, always return a Dataset. By default True.
+    load : bool, optional
+        If True, the data is loaded into memory. By default False.
+    **kwargs:
+        Additional keyword arguments that are passed to the `xr.open_dataset`
+        function.
+
+    Returns
+    -------
+    Dict[str, xr.Dataset]
+        dict of xarray.Dataset
+    """
+    ncs = dict()
+    fns = glob.glob(join(root, fn))
+    if "chunks" not in kwargs:  # read lazy by default
+        kwargs.update(chunks="auto")
+    for fn in fns:
+        name = basename(fn).split(".")[0]
```

Comment:
please use `splitext` instead. if there are multiple periods in the file name this will break I think.